types/d3-sankey/index.d.ts

```diff
@@ -348,7 +350,19 @@ export interface SankeyLayout<Data, N extends SankeyExtraProperties, L extends S
      *
      * @param compare Node comparison function.
      */
-    nodeSort(compare: (a: SankeyNode<N, L>, b: SankeyNode<N, L>) => number): this;
+    nodeSort(compare: (a: SankeyNode<N, L>, b: SankeyNode<N, L>) => number | undefined | null): this;
```

Comment:
I updated this to include `null` and `undefined` as per the docs git push origin HEAD